BuildResidentialHPXML/measure.rb

```diff
@@ -4867,9 +4867,17 @@ def self.set_ducts(args, hvac_distribution)
     end
   end
 
-  def self.set_hvac_control(hpxml, args)
+  def self.set_hvac_control(hpxml, args, epw_file, weather)
     return if (args[:heating_system_type] == 'none') && (args[:cooling_system_type] == 'none') && (args[:heat_pump_type] == 'none')
 
+    sim_calendar_year = 2007 # default to TMY
+    if !hpxml.header.sim_calendar_year.nil?
+      sim_calendar_year = hpxml.header.sim_calendar_year
+    end
+    if epw_file.startDateActualYear.is_initialized # AMY
+      sim_calendar_year = epw_file.startDateActualYear.get
+    end
```

Comment:
Is this duplicating logic from somewhere? Can we have all similar logic in one place?